### PR TITLE
Morph CLI for indexing and searching MCI

### PR DESCRIPTION
**Pull request by aphexcx**
Implemented in rift-engine/rift/ir/morph.py
`./morph index` to index and embed
`./morph search` to search

Currently improving performance by parsing and embedding in parallel so it works faster on larger repos

### CODE REVIEW

File: rift-engine/rift/ir/index.py [hunk 1]
```diff
@@ -158,7 +159,9 @@ def similarity(self, query: Query) -> float:
         Computes the maximum cosine similarity between the vectors in this embedding and the vectors in the given query embedding.
         """
 
-        similarities = [query.node.node_similarity(self.symbol, v) for v in self.vectors]
+        similarities = [
```
Comment (cristianoc) [inline rift-engine/rift/ir/index.py:162]: Formatting:
```
isort --line-length 100 --recursive ./rift-engine && black --line-length 100 ./rift-engine
```

File: rift-engine/rift/ir/morph.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import time
+from typing import List
+
+from index import Index, Query, Text
+import rift.ir.IR as IR
+from rift.ir.parser import parse_files_in_paths
+
+import git
+
+repo_root = git.Repo(".", search_parent_directories=True).git.rev_parse(
```
Comment (cristianoc) [inline rift-engine/rift/ir/morph.py:15]: Do we envisage wanting to index something that is not the root of a repo?

Comment (cristianoc) [reply rift-engine/rift/ir/morph.py:15]: Perhaps some command-line option to select in general what one wants to index.

**Review by cristianoc — COMMENTED** [review]
Looks great.
Would you mind rebasing on top of the fixes here: https://github.com/morph-labs/rift/pull/215

Notice the function `documents_for_symbol` has changed a bit and needs to be copied over again.

**Comment by cristianoc** [comment]
In fact, no need to pass `documents_for_symbol` anymore: https://github.com/morph-labs/rift/pull/203/commits/c4fc635bbc7a6e08a2abdda62c20241d7a8a07c1
The defaults are fine.

**Review by cristianoc — APPROVED** [review]
Great

File: rift-engine/rift/ir/morph.py (hunk 2 above)
Comment (aphexcx) [reply rift-engine/rift/ir/morph.py:15]: Not a bad idea. I think the root of a repo is a sensible default - @jesse-michael-han do you see a usecase for indexing only subdirectories of a repo?

**Comment by cristianoc** [comment]
This has moved

**Pull request closed without merging**